### PR TITLE
Convert SC_UNAUTHORIZED (401)   return codes to SC_FORBIDDEN (403)

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
re: issues 551
supercedes pr: 552

1. Convert all generation of SC_UNAUTHORIZED (401)
   return codes to SC_FORBIDDEN (403).
2. Fix tests to accept SC_FORBIDDEN as well as SC_UNAUTHORIZED

I think issue 551 is still outstanding.

### CODE REVIEW

**Comment by cwardgar** [comment]
Many of the `printf` statements are wrong, e.g.
```java
System.out.printf("Should return HttpStatus.SC_UNAUTHORIZED|HttpStatus.SC_FORBIDDEN err=%s%n", e.getMessage());
```
should be:
```java
System.out.printf("Should return %s or %s; err = %s%n", HttpStatus.SC_UNAUTHORIZED, HttpStatus.SC_FORBIDDEN, e.getMessage());
```

Even better would be to convert those STDOUT messages to logger messages.

**Comment by DennisHeimbigner** [comment]
No, those printf's are correct. We only have a set of possible accepted
return values so we can only list the set since no element of
the set was returned.

You are correct about the logger; let me fix that.
=Dennis

**Comment by cwardgar** [comment]
So in the log, you want to see:
```
Should return HttpStatus.SC_UNAUTHORIZED or HttpStatus.SC_FORBIDDEN; err = <MESSAGE>
```
instead of:
```
Should return 401 or 403; err = <MESSAGE>
```
It's fine if you do; just making sure.

Also, slf4j uses a different formatting anchor than printf. See https://www.slf4j.org/api/org/slf4j/helpers/MessageFormatter.html

File: it/src/test/java/thredds/tds/TestRestrictDataset.java [hunk 1]
```diff
@@ -184,12 +175,12 @@ public void testSuccess() {
...
     String endpoint = TestWithLocalServer.withPath(path);
-    System.out.printf("testRestriction req = '%s'%n", endpoint);
+    logger.info("testRestriction req = '%s'", endpoint);
     try {
       try (HTTPMethod method = HTTPFactory.Get(endpoint)) {
         int statusCode = method.execute();
-        if (statusCode != 401 && statusCode != 403) {
-          System.out.printf("statuscode=%d expected 401 or 403%n", statusCode);
+        if (statusCode != HttpStatus.SC_UNAUTHORIZED && statusCode != HttpStatus.SC_FORBIDDEN) {
+          logger.error("statuscode=%d expected HttpStatus.SC_UNAUTHORIZED or HttpStatus.SC_FORBIDDEN; actual=%d", statusCode);
```
Comment (cwardgar) [inline it/src/test/java/thredds/tds/TestRestrictDataset.java]: Format contains two anchors but only one value is given.

**Comment by DennisHeimbigner** [comment]
That will teach me to do a blanket replace.
=Dennis

**Comment by cwardgar** [comment]
Looks good. Does this address all three of #547, #552, and #625?

**Comment by DennisHeimbigner** [comment]
We need to apply 552, but to same base as this (5.0.0).

**Pull request merged**